src/case_mapping.rs

```diff
@@ -0,0 +1,66 @@
+use std::collections::BTreeMap;
+
+use ucd_parse::{SpecialCaseMapping, UcdFile, UnicodeData};
+
+use args::ArgMatches;
+use error::Result;
+
+pub fn command(args: ArgMatches) -> Result<()> {
+    let dir = args.ucd_dir()?;
+    let mut lower_map: BTreeMap<u32, Vec<u32>> = BTreeMap::new();
+    let mut upper_map: BTreeMap<u32, Vec<u32>> = BTreeMap::new();
+    let mut title_map: BTreeMap<u32, Vec<u32>> = BTreeMap::new();
+        let mut wtr = args.writer("case_mapping")?;
+    for item in UnicodeData::from_dir(dir)? {
+        let item = item?;
+        if let Some(lower) = item.simple_lowercase_mapping {
+            lower_map.insert(item.codepoint.value(), vec![lower.value()]);
+        }
+        if let Some(upper) = item.simple_uppercase_mapping {
+            upper_map.insert(item.codepoint.value(), vec![upper.value()]);
+        }
+        if let Some(title) = item.simple_titlecase_mapping {
+            title_map.insert(item.codepoint.value(), vec![title.value()]);
+        }
+    }
```

Comment:
Is there any particular reason why you're using `UnicodeData.txt` here instead of `CaseFolding.txt`? AIUI, the latter is derived from both `UnicodeData.txt` and `SpecialCase.txt` so it should wind up being simpler.

(It's been a while since I've looked at this, so I might be missing an important bit that you gain by avoiding `CaseFolding.txt`.)

Reply:
Answering my own question: this is trying to map between lower, upper and title case. Where as case folding is a way of normalizing case entirely. If you jumped directly to `CaseFolding.txt`, then you would lose the "lower," "upper" and "title" designations.